api/plugin.go

```diff
@@ -0,0 +1,242 @@
+package api
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	gcommon "github.com/ethereum/go-ethereum/common"
+	gtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/hibiken/asynq"
+	"github.com/labstack/echo/v4"
+	"github.com/vultisig/vultisigner/common"
+	"github.com/vultisig/vultisigner/internal/tasks"
+	"github.com/vultisig/vultisigner/internal/types"
+)
+
+// ERC20 transfer method ABI
+const erc20ABI = `[{
+    "name": "transfer",
+    "type": "function",
+    "inputs": [
+        {"name": "recipient", "type": "address"},
+        {"name": "amount", "type": "uint256"}
+    ],
+    "outputs": [{"name": "", "type": "bool"}]
+}]`
+
+func (s *Server) SignPluginMessages(c echo.Context) error {
+	var req types.PluginKeysignRequest
```

Comment:
Currently , we are using https://github.com/vultisig/commondata/blob/main/proto/vultisig/keysign/v1/keysign_message.proto#L24 to share messages between devices , Maybe plugin should use the same as well? 
For those using DEFI contract , all the contract related stuff can be put into `memo` field

Reply:
Tentatively yes, that's the plan - I was looking at the proto stuff in WalletCore, but haven't used it yet. Added to the list

Reply:
Refer to https://github.com/vultisig/VultiServer/blob/main/chainhelper/utxo_helper.go , I did stat to do some about it before